src/compiler/checker.ts

```diff
@@ -23508,6 +23508,15 @@ namespace ts {
         }
 
         function getTypeReferenceSerializationKind(typeName: EntityName, location?: Node): TypeReferenceSerializationKind {
+            // ensure both `typeName` and `location` are parse tree nodes.
+            typeName = getParseTreeNode(typeName, isEntityName);

```

Comment:
This seems needlessly conservative - `resolveEntityName` does not use parent pointers on the `name` argument if `location` is present - it just draws the names to lookup from it. I don't think it strictly needs to be a parse tree node (and more importantly, couldn't a transform morph one name into another? Shouldn't that be resolved, so the serialized kind is also reflects the transformation?)?

Probably too much of an edge case to care about until someone asks for it, anyway.

Reply:
I'd rather guard against it in any event.